File: Semestrul 4/MPP2/Problema8SC_CSharp/Model/AgeGroup.cs
namespace Problema8SC_CSharp.Model;

public class AgeGroup: Entity<long>
{
    public long Id
    {
        get => GetId();
        set => SetId(value);
    }
    public String Name { get; set; }
    public int MinAge { get; set; }
    public int MaxAge { get; set; }
    
    public AgeGroup(long id, String name, int minAge, int maxAge)
    {
        Id = id;
        Name = name;
        MinAge = minAge;
        MaxAge = maxAge;
    }
    
    public AgeGroup() {}
    
    public AgeGroup(String name, int minAge, int maxAge) : this(0, name, minAge, maxAge)
    {
    }

    public Tuple<int, int> GetAgeRange()
    {
        return new Tuple<int, int>(MinAge, MaxAge);
    }
    
    public override int GetHashCode()
    {
        return Name.GetHashCode();
    }

    public override bool Equals(object? obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;

        AgeGroup other = (AgeGroup)obj;
        return Name == other.Name && MinAge == other.MinAge && MaxAge == other.MaxAge;
    }

    public override String ToString()
    {
        return $"Name: {Name}, MinAge: {MinAge}, MaxAge: {MaxAge}";
    }
}

File: Semestrul 4/MPP2/Client-Server_2Limbaje/Problema8SC/Problema8SC/Problema8Model/src/main/java/problema8/model/AgeGroup.java
package problema8.model;

import java.util.Objects;

public class AgeGroup extends Entity<Long> {
    private String Name;
    private int MinAge;
    private int MaxAge;

    public AgeGroup(Long id, String name, int minAge, int maxAge) {
        setId(id);
        this.Name = name;
        this.MinAge = minAge;
        this.MaxAge = maxAge;
    }

    public AgeGroup(String name, int minAge, int maxAge) {
        this.Name = name;
        this.MinAge = minAge;
        this.MaxAge = maxAge;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        this.Name = name;
    }

    public int getMinAge() {
        return MinAge;
    }

    public void setMinAge(int minAge) {
        this.MinAge = minAge;
    }

    public int getMaxAge() {
        return MaxAge;
    }

    public void setMaxAge(int maxAge) {
        this.MaxAge = maxAge;
    }

    public Tuple<Integer, Integer> getAgeRange() {
        return new Tuple<>(MinAge, MaxAge);
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AgeGroup ageGroup = (AgeGroup) o;
        return MinAge == ageGroup.MinAge && MaxAge == ageGroup.MaxAge && Objects.equals(Name, ageGroup.Name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(Name, MinAge, MaxAge);
    }

    @Override
    public String toString() {
        return "AgeGroup{" +
                "name='" + Name + '\'' +
                ", minAge=" + MinAge +
                ", maxAge=" + MaxAge +
                '}';
    }
}
